baseapp/abci_test.go

```diff
@@ -132,6 +132,27 @@ func TestGetBlockRentionHeight(t *testing.T) {
 	}
 }
 
+// Verifies that the Commiter is called with the checkState.
+func TestCommiterCalledWithCheckState(t *testing.T) {
+	t.Parallel()
+
+	logger := defaultLogger()
+	db := dbm.NewMemDB()
+	name := t.Name()
+	app := NewBaseApp(name, logger, db, nil)
+
+	wasCommiterCalled := false
+	app.commiter = func(ctx sdk.Context) {
+		require.Equal(t, app.checkState.ctx, ctx)
+		wasCommiterCalled = true
+	}
+
+	app.BeginBlock(abci.RequestBeginBlock{Header: tmproto.Header{Height: 1}})
```

Comment:
is this needed to reset deliverState?

Reply:
I wondered the same thing, but the test fails with a nil pointer deference [here](https://github.com/dydxprotocol/cosmos-sdk/blob/7acbfab60ee2d5fa4c7d43f2a84706cd6816578d/baseapp/abci.go#L368) when `Commit` tries to reference `deliverState.ctx`, which is `nil` (it's set by `BeginBlock` using the block header [here](https://github.com/dydxprotocol/cosmos-sdk/blob/7acbfab60ee2d5fa4c7d43f2a84706cd6816578d/baseapp/abci.go#L165)).